src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const answer: GroupsMap<T> = Object.fromEntries(
+    items.map((item) => [item[key] as string, []]),
```

Comment:
Casting `item[key]` to a string using `as string` can be unsafe if `T` has keys that are not strings or if the values under those keys are not actually strings. You should ensure that the function's contract guarantees that the key will always be a string or handle cases where it might not be.